drivers/temperature/adt7420/adt7420.c

```diff
@@ -75,56 +73,84 @@ const struct adt7420_chip_info chip_info[] = {
  *
  * @param dev              - The device structure.
  * @param register_address - Address of the register.
+ * @param data - Pointer to the register value
  *
  * @return register_value  - Value of the register.
 *******************************************************************************/
-uint16_t adt7420_get_register_value(struct adt7420_dev *dev,
-				    uint8_t register_address)
+int adt7420_reg_read(struct adt7420_dev *dev,
+		     uint8_t register_address, uint16_t *data)
 {
 	if (adt7420_is_spi(dev))
-		return adt7420_spi_get_register_value(dev, register_address);
+		return adt7420_spi_reg_read(dev, register_address, data);
 	else
-		return adt7420_i2c_get_register_value(dev, register_address);
+		return adt7420_i2c_reg_read(dev, register_address, data);
 }
 
 /***************************************************************************//**
  * @brief Sets the value of a register SPI/I2C.
  *
  * @param dev              - The device structure.
  * @param register_address - Address of the register.
- * @param num_data_bytes   - Number of bytes.
- * @param data             - Data to be written in input register.
  * @param mask		   - Bit Mask of the bit to be written
  * @param value		   - Value of the bit
  *
  * @return 0 in case of success, -1 otherwise.
 *******************************************************************************/
-int32_t set_register_value(struct adt7420_dev *dev,
-			   uint8_t register_address,
-			   uint8_t num_data_bytes,
-			   uint8_t *data, uint8_t mask, uint8_t value)
+int adt7420_reg_update_bits(struct adt7420_dev *dev,
+			    uint8_t register_address, uint8_t mask, uint8_t value)
 {
 
 	uint8_t data_buffer[3] = {0, 0, 0};
+	uint16_t regval;
+
+	if(adt7420_reg_read(dev, register_address, &regval) != 0)
```

Comment:
`if(adt7420_reg_read(dev, register_address, &regval))`

Reply:
btw, between, `if` and `(adt7420_reg_read(dev, register_address, &regval))` an empty space is required.